lib/mixlib/install/cli.rb

```diff
@@ -36,6 +36,10 @@ def list_versions(product_name, channel)
         default: "x86_64",
         aliases: ["-a"],
         enum: Mixlib::Install::Options::SUPPORTED_ARCHITECTURES.map(&:to_s)
+      option :platform_version_compat,
+        desc: "Enable or disable platform version compatibility mode when using platform detection",
+        type: :boolean,
+        default: true
```

Comment:
It's weird to add an alias for a boolean that defaults to true. I believe it will be very rare that `--no-platform-compat` will be used. I don't think it makes sense in this case.